src/checks/fastlyKeyExpiration.check.js

```diff
@@ -0,0 +1,109 @@
+const fetch = require('node-fetch');
+const moment = require('moment');
+const logger = require('@financial-times/n-logger').default;
+const Check = require('./check');
+const status = require('./status');
+
+const fastlyApiEndpoint = 'https://api.fastly.com/tokens/self';
+/**
+ * @description Polls the current state of a Fastly key expiration date
+ * alert if the key expires in the next week or two
+ * alert if the key has a null expiry date
+ */
+class FastlyKeyExpirationCheck extends Check {
+	constructor(options) {
+		super(options);
+		this.fastlyKey = options.fastlyKey;
+		this.states = {
+			PENDING: {
+				status: status.PENDING,
+				checkOutput: 'Fastly key check has not yet run',
+				severity: this.severity
```

Comment:
**question**: should these not be set to `options.severity` or just hardcoded instead? else if you have a tick with the `FAILED_URGENT_VALIDATION` state then all future states will be set to `1` regardless of `options.severity`.

Reply:
I am going to use Object.freeze for them 😉

Reply:
Sorry, I was wrong, the states' severities are all set in the constructor so it doesn't matter that `this.serverity` can mutate later. Good idea putting `Object.freeze` in to enforce it though!